include/openthread/platform/infra_if.h

```diff
@@ -162,6 +178,22 @@ extern void otPlatInfraIfDiscoverNat64PrefixDone(otInstance        *aInstance,
                                                  uint32_t           aInfraIfIndex,
                                                  const otIp6Prefix *aIp6Prefix);
 
+/**
+ * Get the link-layer address of the infrastructure interface.
+ *
+ * OpenThread invokes this method when the address is required, for example: when generating an ND6 message
+ * which includes a source link-layer address option.
+ *
+ * @param[in]  aInstance                    The OpenThread instance structure.
+ * @param[in]  aInfraIfIndex                The index of the infrastructure interface.
+ * @param[out] aInfraIfLinkLayerAddress     A pointer to infrastructure interface link-layer address.
```

Comment:
```suggestion
 * @param[out] aInfraIfLinkLayerAddress     A pointer to infrastructure interface link-layer address (EUI-48 format).
```

Reply:
Hi, thanks for your suggestion. How about changing the struct `aInfraIfLinkLayerAddress` as Abtink suggested here https://github.com/openthread/openthread/pull/10647#discussion_r1735054941, I think this will be more flexible.

Reply:
That's fine! When we have variable length it still would be good to mention in the API comments the examples of EUI-48 address with length=6, and EUI-64 address with length=8.